packages/vite-plugin-cloudflare/playground/react-spa/__tests__/experimental-headers-and-redirects/assets.spec.ts

```diff
@@ -0,0 +1,120 @@
+				const response = await fetch(viteTestUrl);
+				expect(response.headers.get("X-Header")).toBe("Custom-Value!!!");
+			});
+
+			// Since Vite will return static assets immediately without invoking the Worker at all
+			// such requests will not have _headers applied.
+			failsIf(!isBuild)("applies _headers to static assets", async ({}) => {
+				const response = await fetch(`${viteTestUrl}/vite.svg`);
+				expect(response.headers.get("X-Header")).toBe("Custom-Value!!!");
+			});
+		});
+
+		// All these tests will fail without experimental support turned on
+		describe("_redirects", () => {
+			test("applies _redirects to HTML responses", async ({}) => {
+				const response = await fetch(`${viteTestUrl}/foo`, {
+					redirect: "manual",
+				});
+				expect(response.status).toBe(302);
+				expect(response.headers.get("Location")).toBe("/bar");
+			});
+
+			// Since Vite will return static assets immediately without invoking the Worker at all
+			// such requests will not have _redirects applied.
+			failsIf(!isBuild)("applies _redirects to static assets", async ({}) => {
+				const response = await fetch(`${viteTestUrl}/redirect.svg`, {
+					redirect: "manual",
+				});
+				expect(response.status).toBe(302);
+				expect(response.headers.get("Location")).toBe("/target.svg");
+			});
+
+			// Since Vite will return static assets immediately without invoking the Worker at all
+			// such requests will not have _redirects applied.
+			failsIf(!isBuild)(
+				"supports proxying static assets to rewritten contents with _redirects",
+				async ({}) => {
+					const response = await fetch(`${viteTestUrl}/rewrite.svg`);
+					expect(response.status).toBe(200);
+					expect(await response.text()).toContain("target.svg");
+				}
+			);
+		});
+	}
+);
+
+describe("reloading the server", () => {
+	test.skipIf(isBuild)(
+		"reloads config when the _headers or _redirects files change",
+		async ({ onTestFinished }) => {
+			const headersPath = join(__dirname, "../../public/_headers");
+			const originalHeaders = readFileSync(headersPath, "utf8");
+			const redirectsPath = join(__dirname, "../../public/_redirects");
+			const originalRedirects = readFileSync(redirectsPath, "utf8");
+			onTestFinished(async () => {
+				writeFileSync(headersPath, originalHeaders);
+				writeFileSync(redirectsPath, originalRedirects);
+			});
+
+			const headersBefore = await fetch(viteTestUrl);
+			expect(headersBefore.headers.get("X-Header")).toBe("Custom-Value!!!");
+			const redirectBefore = await fetch(`${viteTestUrl}/foo`, {
+				redirect: "manual",
+			});
+			expect(redirectBefore.status).toBe(302);
+
+			// We make both these changes at the same time because there is something strange about the test setup
+			// where fetches result in 500 errors, due to Miniflare stubs being reused after disposal.
+			writeFileSync(headersPath, "");
+			writeFileSync(redirectsPath, "");
+
+			// Wait for Vite to reload
+			await setTimeout(500);
+
```

Comment:
A little concerned that this could lead to the tests being quite flaky.

Reply:
Ahah! There is a rather nice `expect.poll()` helper!

Reply:
5430df921ca18ed5497f4b19ec58307f08d89ace